rust/kernel/print.rs

```diff
@@ -53,6 +53,7 @@ unsafe fn rust_fmt_argument(buf: *mut c_char, end: *mut c_char, ptr: *const c_vo
         buf: buf as _,
         end: end as _,
     };
+    // SAFETY: the caller must guarantee that `ptr` is valid.
```

Comment:
Since this is an `unsafe` function, we can add this prerequisite to the function docs (in a `Safety` section), even if the function is not public; and then you can say here "`ptr` is valid due to the safety contract`. You should also mention that `ptr` has to be an `Arguments`.

Reply:
Documented the safety contract, though I'm not entirely sure it is correct